synapse_invite_checker/invite_checker.py

```diff
@@ -404,44 +409,100 @@ async def is_domain_allowed(self, domain: str) -> bool:
     async def is_domain_insurance(self, domain: str) -> bool:
         return await self._domain_list_check(lambda fl: fl.is_insurance(domain))
 
+    async def on_create_room(
+        self,
+        requester: Requester,
+        request_content: dict[str, Any],
+        is_request_admin: bool,
+    ) -> None:
+        """
+        Raise a SynapseError if creating a room with invites should be denied
+        """
+        # Unlike `user_may_invite()`, `on_create_room()` runs with the inviter being
+        # a local user and the invitee is remote
+        inviter = requester.user
+        assert self.api.is_mine(
+            inviter.to_string()
+        ), "creating a room for a user that wasn't from this server is strange"
+
+        invites_block: list[str] = request_content.get("invite", [])
+        # Per A_25538, only a single additional user may be invited to a room during
+        # creation. See:
+        # https://gemspec.gematik.de/docs/gemSpec/gemSpec_TI-M_Basis/gemSpec_TI-M_Basis_V1.1.1/#A_25538
+        # Interesting potential error here, they display an http error code of 400, but
+        # then say to use "M_FORBIDDEN". Pretty sure that is a typo
+        if len(invites_block) > 1:
+            raise SynapseError(
+                403,
+                "When creating a room, a maximum of one participant can be invited directly",
+                errors.Codes.FORBIDDEN,
+            )
+
+        #  The spec says this is to be checked by the messenger proxy. I believe this
+        #  leaves a dangling room with only the original creator and no real evidence
+        #  that the other party will not be joining. Logically, blocking the room
+        #  creation in the situation where we already know the other party can't/won't
+        #  attend is cleaner
+        for invitee_entry in invites_block:
+            await self._check_invite(inviter.to_string(), invitee_entry)
```

Comment:
Why not do what synapse does?

```suggestion
        for invitee in invite_list:
            res = await self._check_invite(inviter.to_string(), invitee_entry)
            if res != NOT_SPAM:
              raise SynapseError...
```

What you are doing below of making the invite checks potentially raise an exception and then mapping it back is really confusing.

Reply:
removed all related SynapseErrors. But, for the record, we don't want to raise a SynapseError directly through `user_may_invite()`. It would technically work, but borks up the machinery in Synapse that is expecting to keep running other callbacks. Like I said, it would work, but it feels so dirty

Reply:
Yeah, I didn't want you to raise from user_may_invite, what you did now looks good :)

Reply:
Alright. How about that?